### DISCUSSION TITLE
Check for memory leaks

### DISCUSSION THREAD

**Discussion by matttbe** [linux-netdev/nipa#15] [Ideas]
After the recent sync with Linus tree, our CI at MPTCP started to complain about memory leaks.

There is already one [patch](https://lore.kernel.org/r/20240215213944.508142048@goodmis.org) fixing that. I wanted to report this to you, thinking that it would affect debug builds running with `KMemLeak`, but it looks like leaks are not checked.

Would it be OK to look for that? By doing something like:

```bash
if [ ! -f "/sys/kernel/debug/kmemleak" ]; then
    return 0
fi

echo scan > /sys/kernel/debug/kmemleak
cat /sys/kernel/debug/kmemleak > "${KMEMLEAK}"
if [ -s "${KMEMLEAK}" ]; then
    ./scripts/decode_stacktrace.sh (...) < "${KMEMLEAK}"
    <mark as fail>
    return 1
fi
```

**Comment by kuba-moo** [comment]
More than OK, it'd be great to add! 
We just need to make sure to run the stack traces thru something like `crash_finger_print()` and filter out known ones that come from the Big Penguin's tree.

**Comment by matttbe** [comment]
KMemLeak support has been added by @kuba-moo (thanks!) in https://github.com/linux-netdev/nipa/commit/be5f03c6fbeba90bd733986a72d35b9cd6c30a63. We can then close this discussion.
